src/mscorlib/src/System/Environment.cs

```diff
@@ -672,19 +614,66 @@ private static string GetEnvironmentVariableCore(string variable, EnvironmentVar
 #endif
         }
 
-        private static IDictionary GetEnvironmentVariablesCore()
+        internal static IEnumerable<KeyValuePair<string, string>> EnumerateEnvironmentVariables()
         {
-            return GetRawEnvironmentVariables();
+            // Format for GetEnvironmentStrings is:
+            // (=HiddenVar=value\0 | Variable=value\0)* \0
+            // See the description of Environment Blocks in MSDN's
+            // CreateProcess page (null-terminated array of null-terminated strings).
+            // Note the =HiddenVar's aren't always at the beginning.
+
+            // Copy strings out, parsing into pairs and inserting into the table.
+            // The first few environment variable entries start with an '='.
+            // The current working directory of every drive (except for those drives
+            // you haven't cd'ed into in your DOS window) are stored in the 
+            // environment block (as =C:=pwd) and the program's exit code is 
+            // as well (=ExitCode=00000000).
+
```

Comment:
Change the CoreRT implementation to match this one? This one looks easier to follow.

I do not think it will be easy to use Span in the CoreRT one because Span is stack-only and cannot be used in iterators.